Lab1_Operationalizing_Pytorch_with_Mlflow/README.md

```diff
@@ -1,31 +1,127 @@
+- Michel Vasconcelos
+- Varuna Jayasiri
 
-Teams involved in this scenario are:
+## Repo Organization
 
-- Team 1 : Michel Vasconcelos, Alexey Naiden, Varuna Jayasiri, John Savage
-- Team 2 : Itirell, Mark Peters, Carlos Leyson, Semendiak
-- Team 3 : Laszlo Sragner, yartem, Dimi, Paulo Maia
-- Team 4 : Shadab Hussain, susana brandao, Godwin Ekainu, Josh
+Following list describes the most relevant folders in which this lab is organized. Some folders in 
+this repo are self-explanatory (like [_imgs_](imgs/)) and therefore aren'r mentioned in the list.
 
+* [src](src/): Main source code folder. It comprises all software produced during this lab. _Src_ 
+itself is organized into subfolders according to which components it belongs to;
+* [docs](docs/): Binary documentation, drawings and reports produced during this lab;
 
-## Contributing
-If you want to join the initiative, please join the MLOps Community on [Slack](https://mlops-community.slack.com/join/shared_invite)
+## Proposal
+### Design Overview
+
+Our team intend to orchestrate common unix management toolkits (shell commands, ssh, etc), MLFlow
+CLI and MLFlow components to build a seamless end-to-end pipeline. It should comprehend 
+[Continuous Integration](https://en.wikipedia.org/wiki/Continuous_integration) (CI), 
+[Continuous Delivery](https://en.wikipedia.org/wiki/Continuous_delivery) (CD) and 
+[Continuous Traning](https://cloud.google.com/solutions/machine-learning/mlops-continuous-delivery-and-automation-pipelines-in-machine-learning) (CT). 
+The following figure depicts the proposed architecture.
+
+![Proposed architecture](imgs/arch.svg)
+
+**Nodes** are bare-metal or virtual machines that host a part of the pipeline process. **ML components** 
+are ML-based code that should be trained, tested and deployed by end-to-end pipeline.They must be 
+packaged as [MLFlow projects](https://mlflow.org/docs/latest/projects.html). **Training nodes**
+host the training process which runs in **Training containers**, running dockerized images built 
+with MLFlow and necessary libs to train ML components. Each ML component defines its needs, that is, 
+libs and external dependencies. To ensure traceability, provenance and model configuration management, 
+one may track hyperparameters, training steps, test results, extra data and the model itself into 
+_MLFlow Tracking Server_ hosted in the **Tracking Node**. All these items, metadata and trained models, 
+are stored in external resources: RDBMS and Cloud Storage Services. Trained models are embedded into 
+dockerized **Serving Instances**, containers running [Torchserve](https://pytorch.org/serve/) providing 
+web access to the models.
+
+In order to provide feedback loop, a **Monitor Node** checks the status and performance meters, issuing 
+new training cycles or a full rebuild process. Some structures traverses through Development and Production perimeters. Training and Monitor nodes are key elements that communicate between those zones. 
```

Comment:
> Some structures traverses through Development and Production perimeters

I don't get this phrase. Would you care to elaborate?

Reply:
Usually, when we think about "system architecture" we try to isolate elements in order to comply with security requirements. Something like Development traffic never reaches Production zone, that is, Development nodes would "lie" in Development zone with no connectivity with nodes \ servers in Production. However, when I drew that, I was thinking a "peculiar" Monitoring Node. One that could reach Development and Production elements. It would lie in between zones. Just like a "cross-cutting concern". 

I was trying to (concisely) say that with that phrase. However, my summarization skills are flawed somehow. hahahahahaahahaha 

In fact, maybe we shouldn't explain that at all. Just remove that keep it simple. What do you think? :-)

Reply:
I have removed it. ;-)